plugins/csp-recorder/src/Plugin.cpp

```diff
@@ -0,0 +1,238 @@
+
+  // Add the record-button to the timeline. We remove / add the button every time it's pressed in
+  // order to change its icon from a circle to a square and back again.
+  mGuiManager->addTimelineButton(
+      "Start Recording", "fiber_manual_record", "recorder.toggleRecording");
+
+  mGuiManager->getGui()->registerCallback(
+      "recorder.toggleRecording", "Enables or disables recording.", std::function([this]() {
+        if (mRecording) {
+          mGuiManager->removeTimelineButton("Stop Recording");
+          mGuiManager->addTimelineButton(
+              "Start Recording", "fiber_manual_record", "recorder.toggleRecording");
+          mRecording = false;
+        } else {
+          mGuiManager->removeTimelineButton("Start Recording");
+          mGuiManager->addTimelineButton("Stop Recording", "stop", "recorder.toggleRecording");
+          mRecording = true;
+        }
+      }));
+
+  // Add a callback to toggle recording of the observer transformation.
+  mGuiManager->getGui()->registerCallback("recorder.setRecordObserver",
+      "Enables or disables recording of the observer transformation.",
+      std::function([this](bool value) { mPluginSettings.mRecordObserver = value; }));
+  mPluginSettings.mRecordObserver.connectAndTouch(
+      [this](bool enable) { mGuiManager->setCheckboxValue("recorder.setRecordObserver", enable); });
+
+  // Add a callback to toggle recording of the simulation time.
+  mGuiManager->getGui()->registerCallback("recorder.setRecordTime",
+      "Enables or disables recording of the current simulation time.",
+      std::function([this](bool value) { mPluginSettings.mRecordTime = value; }));
+  mPluginSettings.mRecordTime.connectAndTouch(
+      [this](bool enable) { mGuiManager->setCheckboxValue("recorder.setRecordTime", enable); });
+
+  // Add a callback to toggle recording of the HDR exposure.
+  mGuiManager->getGui()->registerCallback("recorder.setRecordExposure",
+      "Enables or disables recording of the current camera exposure.",
+      std::function([this](bool value) { mPluginSettings.mRecordExposure = value; }));
+  mPluginSettings.mRecordExposure.connectAndTouch(
+      [this](bool enable) { mGuiManager->setCheckboxValue("recorder.setRecordExposure", enable); });
+
+  // Load initial settings.
+  onLoad();
+
+  logger().info("Loading done.");
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void Plugin::update() {
+
+  if (mRecording) {
+
+    // We are recording but haven't opened the output file - that means it's the very first frame of
+    // the current recording session. So open the file!
+    if (!mOutFile.is_open()) {
+
+      // Reset the frame counter.
+      mFrameCounter = 0;
+
+      // We use the current date as a filename.
+      std::string fileName =
+          "recording-" +
+          cs::utils::convert::time::toString(boost::posix_time::microsec_clock::local_time()) +
```

Comment:
the plugin did not create the script on my windows machine. It has some issue with the time string because changing the filename to "recording-bla.py" just for testing worked.

Reply:
Adding `cs::utils::replaceString(fileName, ":", "-");` fixed it.